packages/http-client-csharp/generator/TestProjects/Local/Unbranded-TypeSpec/src/Generated/Internal/ModelSerializationExtensions.cs

```diff
@@ -0,0 +1,284 @@
+
+        /// <param name="element"></param>
+        /// <param name="format"></param>
+        public static byte[] GetBytesFromBase64(this System.Text.Json.JsonElement element, string format)
+        {
+            if (element.ValueKind == System.Text.Json.JsonValueKind.Null)
+            {
+                return null;
+            }
+
+            return format switch
+            {
+                "U" => TypeFormatters.FromBase64UrlString(element.GetRequiredString()),
+                "D" => element.GetBytesFromBase64(),
+                _ => throw new ArgumentException($"Format is not supported: '{format}'", nameof(format))
+            };
+        }
+
+        /// <param name="element"></param>
+        /// <param name="format"></param>
+        public static DateTimeOffset GetDateTimeOffset(this System.Text.Json.JsonElement element, string format) => format switch
+        {
+            "U" when element.ValueKind == System.Text.Json.JsonValueKind.Number => DateTimeOffset.FromUnixTimeSeconds(element.GetInt64()),
+            _ => TypeFormatters.ParseDateTimeOffset(element.GetString(), format)
+        };
+
+        /// <param name="element"></param>
+        /// <param name="format"></param>
+        public static TimeSpan GetTimeSpan(this System.Text.Json.JsonElement element, string format) => TypeFormatters.ParseTimeSpan(element.GetString(), format);
+
+        /// <param name="element"></param>
+        public static char GetChar(this System.Text.Json.JsonElement element)
+        {
+            if (element.ValueKind == System.Text.Json.JsonValueKind.String)
+            {
+                var text = element.GetString();
+                if (text == null || text.Length != 1)
+                {
+                    throw new NotSupportedException($"Cannot convert \"{text}\" to a char");
+                }
+                return text[0];
+            }
+                else
+                {
+                throw new NotSupportedException($"Cannot convert {element.ValueKind} to a char");
+            }
+        }
+
+        /// <param name="property"></param>
+        [Conditional("DEBUG")]
+        public static void ThrowNonNullablePropertyIsNull(this System.Text.Json.JsonProperty @property)
+        {
+            throw new System.Text.Json.JsonException($"A property '{@property.Name}' defined as non-nullable but received as null from the service. This exception only happens in DEBUG builds of the library and would be ignored in the release build");
+        }
+
+        /// <param name="element"></param>
+        public static string GetRequiredString(this System.Text.Json.JsonElement element)
+        {
+            var value = element.GetString();
+            if (value == null)
+            {
+                throw new InvalidOperationException($"The requested operation requires an element of type 'String', but the target element has type '{element.ValueKind}'.");
+            }
+            return value;
+        }
+
```

Comment:
Should we generate minimal description?

Reply:
These are internal so not needed but if you think its a custom code usability thing lets open a tracking issue and we can address it when we do the custom code work.